src/main/scala/com/databricks/spark/sql/perf/tpcds/TPCDS_Standalone.scala

```diff
@@ -0,0 +1,276 @@
+        .action((x, c) => c.copy(regenerateDataset = x))
+        .text("whether to regenerate the test dataset")
+      opt[Long]('c', "compare")
+        .action((x, c) => c.copy(baseline = Some(x)))
+        .text("the timestamp of the baseline experiment to compare with")
+      help("help")
+        .text("prints this usage text")
+    }
+
+    parser.parse(args, TpcdsStandaloneConfig()) match {
+      case Some(config) =>
+        run(config)
+      case None =>
+        System.exit(1)
+    }
+  }
+
+  def run(conf: TpcdsStandaloneConfig): Unit = {
+    val spark = SparkSession
+      .builder()
+      .master(conf.sparkMaster)
+      .appName(getClass.getName)
+      .getOrCreate()
+
+    logger.info(s"=== DATASET LOCATION: ${generateDatasetLocation(conf)}")
+    logger.info(s"=== DB NAME: ${generateDbName(conf)}")
+
+    if (conf.regenerateDataset || !checkDirExists(generateDatasetLocation(conf), new Configuration())) {
+      logger.info("=== GENERATING TEST DATASET ===")
+      generateDataset(spark, conf)
+      logger.info("=== TEST DATASET GENERATION COMPLETE ===")
+    } else {
+      logger.info("=== TEST DATASET ALREADY GENERATED ===")
+    }
+
+    if (conf.regenerateDataset || !spark.catalog.databaseExists(generateDbName(conf))) {
+      logger.info("=== GENERATING METADATA ===")
+      generateMetadata(spark, conf)
+      logger.info("=== TEST METADATA GENERATION COMPLETE ===")
+    } else {
+      logger.info("=== TEST METADATA ALREADY GENERATED ===")
+    }
+
+    logger.info("=== STARTING EXPERIMENT ===")
+    spark.sql(s"use ${generateDbName(conf)}")
+    executeExperiment(spark, conf)
+    logger.info("=== EXPERIMENT COMPLETE ===")
+  }
+
+  /** Check if the file exists at the given path. */
+  def checkDirExists(path: String, conf: Configuration): Boolean = {
+    val hdpPath = new Path(path)
+    val fs = getFileSystemForPath(hdpPath, conf)
+    fs.exists(hdpPath)
+  }
+
+  def getFileSystemForPath(path: Path, conf: Configuration): FileSystem = {
+    // For local file systems, return the raw local file system, such calls to flush()
+    // actually flushes the stream.
+    val fs = path.getFileSystem(conf)
+    fs match {
+      case localFs: LocalFileSystem => localFs.getRawFileSystem
+      case _ => fs
+    }
+  }
+
+  def executeExperiment(spark: SparkSession, conf: TpcdsStandaloneConfig): Unit = {
+    spark.conf.set("spark.sql.shuffle.partitions", conf.shufflePartitions)
+
+    val tpcds = new TPCDS(sqlContext = spark.sqlContext)
+    val experiment = tpcds.runExperiment(
+      tpcds.tpcds2_4Queries,
+      iterations = conf.iterations,
+      resultLocation = conf.outputLocation,
+      tags = Map(
+        "runtype" -> "benchmark",
+        "database" -> generateDbName(conf),
+        "scale_factor" -> conf.scaleFactor.toString))
+
+    logger.info(experiment.toString)
+    experiment.waitForFinish(conf.timeoutHours * 60 * 60)
+
+    spark.sqlContext.setConf("spark.sql.shuffle.partitions", "1")
```

Comment:
better explicitly .repartition(1) or .coalesce(1) where you need it? (and you seem to already be doing that)